src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+import { Cache, caching } from 'cache-manager';
+import { getConn, IPrefixCommand, IPrefixCommandCategory, IPrefixCommandChannelDefaultVersion, IPrefixCommandVersion, Logger, PrefixCommand, PrefixCommandCategory, PrefixCommandChannelDefaultVersion, PrefixCommandVersion } from '../index';
+
+let inMemoryCache: Cache;
+const cacheSize = 10000;
+const cacheRefreshInterval = process.env.CACHE_REFRESH_INTERVAL ? Number(process.env.CACHE_REFRESH_INTERVAL) : 1800;
+const cacheTTL = cacheRefreshInterval * 2 * 1000;
+
+/**
+ * Cache Prefixes
+ */
+
+export const memoryCachePrefixCommand = 'PF_COMMAND';
+export const memoryCachePrefixVersion = 'PF_VERSION';
+export const memoryCachePrefixCategory = 'PF_CATEGORY';
+export const memoryCachePrefixChannelDefaultVersion = 'PF_CHANNEL_VERSION';
+
+/**
+ * Cache Management Functions
+ */
+
+export async function setupInMemoryCache(callback = Logger.error) {
+    try {
+        inMemoryCache = await caching(
+            'memory',
+            {
+                ttl: cacheTTL,
+                max: cacheSize,
+            },
+        );
+        Logger.info('In memory cache set up');
+    } catch (err) {
+        callback(err);
+    }
+}
+
+export function getInMemoryCache(callback = Logger.error) {
+    if (!inMemoryCache) {
+        callback(new Error('No in memory cache available.'));
+        return null;
+    }
+    return inMemoryCache;
+}
+
+/**
+ * Prefix Command Cache Management Functions
+ */
+
+export async function clearSinglePrefixCommandCache(command: IPrefixCommand) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name, aliases } = command;
+    Logger.debug(`Clearing cache for command or alias "${name}"`);
+    for (const alias of aliases) {
+        // eslint-disable-next-line no-await-in-loop
+        await inMemoryCache.del(`${memoryCachePrefixCommand}:${alias.toLowerCase()}`);
+    }
+    await inMemoryCache.del(`${memoryCachePrefixCommand}:${name.toLowerCase()}`);
+}
+
+export async function loadSinglePrefixCommandToCache(command: IPrefixCommand) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { name, aliases } = command;
+    Logger.debug(`Loading command ${name} to cache`);
+    await inMemoryCache.set(`${memoryCachePrefixCommand}:${name.toLowerCase()}`, command.toObject());
+    for (const alias of aliases) {
+        // eslint-disable-next-line no-await-in-loop
+        await inMemoryCache.set(`${memoryCachePrefixCommand}:${alias.toLowerCase()}`, command.toObject());
+    }
+}
+
+export async function loadAllPrefixCommandsToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommands = await PrefixCommand.find();
+    for (const command of PrefixCommands) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandToCache(command);
+    }
```

Comment:
```suggestion
    await Promise.all(PrefixCommands.map((command) => loadSinglePrefixCommandToCache(command)));
```
`Promise.all()` is cool :D